Complement description in section of Mutex drivers in readme (#38)
yiisoft/mutex: README.md

There is ready to extend `MutexFactory` and a `RetryAcquireTrait` that contains `retryAcquire()` method implementing
the "wait for a lock for a certain time" functionality.

> When implementing your own drivers, you need to take care of automatic unlocking. For example using a destructor:

Reviewer: I'd mention shutdown function as well.

Author: Done.

```php
public function __destruct()
{
    $this->release();
}
```

> Note that you do not need to call the `exit()` or `die()` functions in the destructor. Since calling

Reviewer: ```suggestion
Note that you should not call the `exit()` or `die()` functions in the destructor. Since calling
```
